pkgs/html/lib/src/html_input_stream.dart

```diff
@@ -207,16 +221,31 @@ class HtmlInputStream {
   /// EOF when EOF is reached.
   String? char() {
     if (_offset >= _chars.length) return eof;
-    return _isSurrogatePair(_chars, _offset)
-        ? String.fromCharCodes([_chars[_offset++], _chars[_offset++]])
-        : String.fromCharCode(_chars[_offset++]);
+    final firstCharCode = _chars[_offset++];
+    if (firstCharCode < 256) {
+      return asciiCharacters[firstCharCode];
+    }
+    if (_isSurrogatePair(_chars, _offset - 1)) {
+      return String.fromCharCodes([firstCharCode, _chars[_offset++]]);
+    }
+    return String.fromCharCode(firstCharCode);
+  }
+
+  int? peekCodeUnit() {
+    if (_offset >= _chars.length) return null;
+    return _chars[_offset];
   }
 
   String? peekChar() {
     if (_offset >= _chars.length) return eof;
-    return _isSurrogatePair(_chars, _offset)
-        ? String.fromCharCodes([_chars[_offset], _chars[_offset + 1]])
-        : String.fromCharCode(_chars[_offset]);
+    final charCode = _chars[_offset];
+    if (charCode < 256) {
+      return asciiCharacters[charCode];
```

Comment:
Does changing this to `String.fromCharCode(charCode)` affect the performance much?

Reply:
The improvement is 1% in AoT compilation and 6% in JIT